File: common/src/main/java/cc/turtl/turtlshell/util/MessageUtils.java
package cc.turtl.turtlshell.util;

import cc.turtl.turtlshell.TurtlShellConstants;
import cc.turtl.turtlshell.util.format.ColorUtils;
import com.mojang.brigadier.context.CommandContext;
import net.minecraft.client.player.LocalPlayer;
import net.minecraft.network.chat.Component;
import org.jetbrains.annotations.NotNull;

import static cc.turtl.turtlshell.util.format.ComponentUtils.createComponent;
import static cc.turtl.turtlshell.util.format.ComponentUtils.labelled;

public class MessageUtils {
    private MessageUtils() {
    }

    public static void executeCommand(@NotNull LocalPlayer player, String command) {
        player.connection.sendCommand(command.startsWith("/") ? command.substring(1) : command);
    }

    public static void sendEmptyLine(@NotNull LocalPlayer player) {
        send(player, Component.empty());
    }

    public static void send(@NotNull LocalPlayer player, Component message) {
        player.sendSystemMessage(message);
    }

    public static void send(@NotNull LocalPlayer player, String message) {
        send(player, Component.literal(message));
    }

    public static void sendPrefixed(@NotNull LocalPlayer player, Component message) {
        send(player, Component.empty().append(TurtlShellConstants.MESSAGE_PREFIX).append(message));
    }

    public static void sendPrefixed(@NotNull LocalPlayer player, String message) {
        sendPrefixed(player, createComponent(message, ColorUtils.WHITE.getRGB()));
    }

    public static void sendSuccess(@NotNull LocalPlayer player, String message) {
        sendPrefixed(player, createComponent(message, ColorUtils.GREEN.getRGB()));
    }

    public static void sendWarning(@NotNull LocalPlayer player, String message) {
        sendPrefixed(player, createComponent(message, ColorUtils.YELLOW.getRGB()));
    }

    public static <S> void sendError(@NotNull LocalPlayer player, CommandContext<S> context, Exception e) {
        sendPrefixed(player, createComponent("Error executing: " + context.getInput(), ColorUtils.RED.getRGB()));
        TurtlShellConstants.LOGGER.error("Error executing '{}': ", context, e);
    }

    public static void sendLabeled(@NotNull LocalPlayer player, String label, Object value) {
        sendPrefixed(player, labelled(label, value));
    }
}

File: common/src/main/kotlin/cc/turtl/turtlshell/util/format/component/MessageUtils.kt
package cc.turtl.turtlshell.util.format.component

import cc.turtl.turtlshell.TurtlShellConstants.MESSAGE_PREFIX
import net.minecraft.network.chat.Component
import net.minecraft.world.entity.player.Player

fun Player.sendEmptyLine() = this.sendSystemMessage(Component.empty())
fun Player.sendPrefixed(message: Component) = this.sendSystemMessage(MESSAGE_PREFIX.append(message))
fun Player.sendPrefixed(message: String) = this.sendPrefixed(componentOf(message))
fun Player.sendSuccess(message: String) = this.sendPrefixed(componentOf(message).green())
fun Player.sendWarning(message: String) = this.sendPrefixed(componentOf(message).yellow())
fun Player.sendError(message: String) = this.sendPrefixed(componentOf(message).red())
fun Player.sendLabelled(label: String, value: Component) = this.sendPrefixed(value.copy().labelled(label))
fun Player.sendLabelled(label: String, value: String) = this.sendLabelled(label, componentOf(value))